### PR TITLE
Translations/ add language selection page for Connect and Grammar

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add language selection page for Connect and Grammar

## WHY
this is part of the larger translation project

## HOW
pull language selection page out of Diagnostic into a shared component, and conditionally render it for Connect and Grammar

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2024-07-30 at 7 40 25 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Add-a-language-picker-landing-page-before-starting-an-activity-that-has-a-translation-7847aec660a4473bb7ff3ec2a48fa510?pvs=4

### What have you done to QA this feature?
(Provide enough detail that a reviewer could assess whether additional QA should be done. For larger projects, additionally use the Engineer Feature Testing Notion template. Review Guidelines if needed: https://www.notion.so/quill/Github-PR-QA-Guidelines-49e99fc965654ceeb8c6249bd9d181d7)
verified locally and on staging that the language page is rendered when showTranslation flag is present, and selection is processed as expected

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes

### CODE REVIEW

**Review by samanthamjohn — APPROVED** [review]
looks great!

File: services/QuillLMS/client/app/bundles/Connect/components/studentLessons/lesson.jsx [hunk 1]
```diff
@@ -525,6 +527,15 @@ export class Lesson extends React.Component {
           saveToLMS={this.saveToLMS}
         />
       );
+    } else if (languageOptions && hasTranslationFlag()) {
```
Comment (emilia-friedberg) [inline services/QuillLMS/client/app/bundles/Connect/components/studentLessons/lesson.jsx]: Are we ever going to see the Register component this way? Maybe it doesn't matter or I'm reading it incorrectly, but it seems like we may never hit the `else` block if this is the case.

**Review by emilia-friedberg — APPROVED** [review]
Looks great! Thanks for adding tests.

File: services/QuillLMS/client/app/bundles/Connect/components/studentLessons/lesson.jsx (hunk 1 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Connect/components/studentLessons/lesson.jsx]: that's a good catch-- I'll need to rework the logic a bit

**Pull request merged**